File: MiniDOMBase/DMD/Anagrafica/Persone/CPersonaComparer.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases



Partial Public Class Anagrafica
     

    Public Class CPersonaComparer
        Implements IComparer

        Public Sub New()
            DMDObject.IncreaseCounter(Me)
        End Sub

        Protected Overrides Sub Finalize()
            MyBase.Finalize()
            DMDObject.DecreaseCounter(Me)
        End Sub

        Public Function Compare(ByVal a As CPersona, ByVal b As CPersona) As Integer
            Return Strings.Compare(a.Nominativo, b.Nominativo)
        End Function

        Private Function Compare1(x As Object, y As Object) As Integer Implements IComparer.Compare
            Return Me.Compare(x, y)
        End Function
    End Class
     

End Class

File: MiniDOMBase/DMD/Anagrafica/Persone/Comparers/CPersonaComparer.cs
using System.Collections;
using System.Collections.Generic;

namespace minidom
{
    public partial class Anagrafica
    {

        /// <summary>
        /// Oggetto comparer tra due persone
        /// </summary>
        public class CPersonaComparer
            : IComparer, IComparer<CPersona>
        {

            /// <summary>
            /// Costruttore
            /// </summary>
            public CPersonaComparer()
            {
                ////DMDObject.IncreaseCounter(this);
            }

            /// <summary>
            /// Distruttore
            /// </summary>
            ~CPersonaComparer()
            {
                ////DMDObject.DecreaseCounter(this);
            }

            /// <summary>
            /// Compare due persone per nominativo
            /// </summary>
            /// <param name="a"></param>
            /// <param name="b"></param>
            /// <returns></returns>
            public virtual int Compare(CPersona a, CPersona b)
            {
                return DMD.Strings.Compare(a.Nominativo, b.Nominativo, true);
            }

            int IComparer.Compare(object x, object y)
            {
                return Compare((CPersona)x, (CPersona)y);
            }
        }
    }
}